third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -766,3 +770,60 @@ impl<'env> ModuleContext<'env> {
         s.display(self.env.symbol_pool()).to_string()
     }
 }
+
+impl<'env> ModuleContext<'env> {
+    /// Acquires analysis. This is temporary until we have the full reference analysis.
+    fn generate_acquires_map(&self, module: &ModuleEnv) -> BTreeMap<FunId, BTreeSet<StructId>> {
+        // Compute map with direct usage of resources
+        let mut usage_map = module
+            .get_functions()
+            .map(|f| (f.get_id(), self.get_direct_function_acquires(&f)))
+            .collect::<BTreeMap<_, _>>();
+        // Now run a fixed-point loop: add resources used by called functions until there are no
+        // changes.
+        loop {
+            let mut changes = false;
+            for fun in module.get_functions() {
+                if let Some(callees) = fun.get_called_functions() {
+                    let mut usage = usage_map[&fun.get_id()].clone();
+                    let count = usage.len();
+                    // Extend usage by that of callees from the same module. Acquires is only
+                    // local to a module.
+                    for callee in callees {
+                        if callee.module_id == module.get_id() {
+                            usage.extend(usage_map[&callee.id].iter().cloned());
+                        }
+                    }
+                    if usage.len() > count {
+                        *usage_map.get_mut(&fun.get_id()).unwrap() = usage;
+                        changes = true;
+                    }
+                }
+            }
+            if !changes {
+                break;
+            }
+        }
+        usage_map
+    }
+
+    fn get_direct_function_acquires(&self, fun: &FunctionEnv) -> BTreeSet<StructId> {
+        let mut result = BTreeSet::new();
+        let target = self.targets.get_target(fun, &FunctionVariant::Baseline);
+        for bc in target.get_bytecode() {
+            use Bytecode::*;
+            use Operation::*;
+            match bc {
+                Call(_, _, MoveFrom(mid, sid, ..), ..)
+                | Call(_, _, BorrowGlobal(mid, sid, ..), ..)
```

Comment:
Its only BorrowGlobal, mutability is derived from the type. 

GetGlobal is a construct of the verification language and not available in the programming language, and mentioned here for robustness because perhaps we use in the p-language as well in the future (it is borrow_global followed by deref). Added a comment to clarify.